buffer/src/main/java/io/netty5/buffer/adapt/AdaptivePoolingAllocator.java

```diff
@@ -165,7 +204,17 @@ public Buffer allocate(int size) {
         InternalBufferUtils.assertValidBufferSize(size);
         if (size <= MAX_CHUNK_SIZE) {
             int sizeBucket = AllocationStatistics.sizeBucket(size); // Compute outside of Magazine lock for better ILP.
-            long threadId = threadId(Thread.currentThread());
+            FastThreadLocal<Object> threadLocalMagazine = this.threadLocalMagazine;
+            Thread currentThread = Thread.currentThread();
+            if (threadLocalMagazine != null && currentThread instanceof FastThreadLocalThread) {
+                // this is going to exclude virtual threads,
+                // but sadly even custom event executor which are not FastThreadLocalThreads
```

Comment:
it wouldn't be the only performance snag for such event loops.

Reply:
So I can drop this comment: in my mind i wanted the event loop to become a long-running virtual thread, TBH...
it would be great if we could make it faster/easier  to spot when the current thread is an executor one